service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -28,61 +33,67 @@
     private static final String MIGRATION_ID_KEY = "migrationId";
     private final GeneratedOrderRemovalAction generatedOrderRemovalAction;
+    private final CMORemovalAction cmoRemovalAction;
 
     @PostMapping("/about-to-submit")
     public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         Object migrationId = caseDetails.getData().get(MIGRATION_ID_KEY);
 
-        if ("FPLA-2684".equals(migrationId)) {
+        if ("FPLA-2693".equals(migrationId)) {
             Object hiddenOrders = caseDetails.getData().get("hiddenOrders");
-            run2684(caseDetails);
+            run2693(caseDetails);
             caseDetails.getData().put("hiddenOrders", hiddenOrders);
         }
 
-        if ("FPLA-2687".equals(migrationId)) {
-            Object hiddenOrders = caseDetails.getData().get("hiddenOrders");
-            run2687(caseDetails);
-            caseDetails.getData().put("hiddenOrders", hiddenOrders);
+        if("FPLA-2716".equals(migrationId)) {
+            run2716(caseDetails);
         }
 
         caseDetails.getData().remove(MIGRATION_ID_KEY);
         return respond(caseDetails);
     }
 
-    private void run2684(CaseDetails caseDetails) {
+    private void run2716(CaseDetails caseDetails) {
         CaseData caseData = getCaseData(caseDetails);
 
-        if ("CF20C50070".equals(caseData.getFamilyManCaseNumber())) {
-            CaseDetailsMap caseDetailsMap = CaseDetailsMap.caseDetailsMap(caseDetails);
-
-            if (caseData.getOrderCollection().size() < 3) {
-                throw new IllegalArgumentException(format("Expected at least three orders but found %s",
-                    caseData.getOrderCollection().size()));
-            }
+        if ("CF21C50009".equals(caseData.getFamilyManCaseNumber())) {
+            removeC21OrderFromFirstHearingOrderBundle(caseDetails);
+        }
+    }
 
-            Element<GeneratedOrder> orderThree = caseData.getOrderCollection().get(2);
+    private void removeC21OrderFromFirstHearingOrderBundle(CaseDetails caseDetails) {
+        CaseData caseData = getCaseData(caseDetails);
 
-            generatedOrderRemovalAction.remove(caseData, caseDetailsMap, orderThree.getId(), orderThree.getValue());
+        List<Element<HearingOrder>> orders = caseData.getHearingOrdersBundlesDrafts().get(0).getValue().getOrders();
 
-            caseDetails.setData(caseDetailsMap);
+        if(orders.size() < 2) {
+            throw new IllegalArgumentException(format("Expected at least 2 orders in hearing order bundle but found %s",
+                orders.size()));
         }
+
+        orders.remove(1);
+
+        List<Element<HearingOrdersBundle>> updatedBundle = caseData.getHearingOrdersBundlesDrafts();
+        updatedBundle.get(0).getValue().setOrders(orders);
+
+        caseDetails.getData().put("hearingOrdersBundlesDrafts", updatedBundle);
```

Comment:
How do we know draft 2 is a C21, did you get the data? Should we check that the order is a C21 in method?